src/connector/src/sink/catalog/mod.rs

```diff
@@ -54,6 +54,54 @@ impl From<SinkId> for u32 {
     }
 }
 
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum SinkType {
+    /// The data written into the sink connector can only be INSERT. No UPDATE or DELETE is
+    /// allowed.
+    AppendOnly,
+    /// The input of the sink operator can be INSERT, UPDATE, or DELETE, but it must drop any
+    /// UPDATE or DELETE and write only INSERT into the sink connector.
+    ForceAppendOnly,
+    /// The data written into the sink connector can be INSERT, UPDATE, or DELETE.
+    Upsert,
+    /// The sink can be either append-only or upsert. We should try append-only first. If
+    /// unavailable, fall back to upsert.
+    Any,
```

Comment:
in which case will SinkType be any

Reply:
When the frontend derivation is append-only and the user also defines the sink as append-only. In this case, the sink **should** be append-only, but it's also correct that it is upsert.